chirp/drivers/hg_uv98.py

```diff
@@ -0,0 +1,897 @@
+        raise errors.RadioError("Radio refused program mode: {}".format(ack))
+    radio.pipe.write(b"\x02")
+    ident = radio.pipe.read(8)
+    LOG.debug('ident string was %r' % ident)
+    if ident != radio.IDENT:
+        raise errors.RadioError(
+            "Incorrect model: {}, expected {!r}".format(
+                util.hexprint(ident), radio.IDENT,
+            )
+        )
+    LOG.info("Model: %s (%s)" % (radio.MODEL, util.hexprint(ident)))
+    radio.pipe.write(b"\x06")
+    ack = radio.pipe.read(1)
+    if ack != b"\x06":
+        raise errors.RadioError("Radio entered program mode, but didn't ack our ack")
+
+
+def do_download(radio):
+    radio.pipe.parity = "E"
+    radio.pipe.timeout = 1
+    do_ident(radio)
+
+    data = bytes(b"")
+    for addr in range(0, MAX_ADDR, CHUNK_SIZE):
+        send(radio, make_frame(bytes(b"R"), addr, CHUNK_SIZE))
+        _addr, _data = recv(radio)
+        if _addr != addr:
+            raise errors.RadioError("Radio sent unexpected address")
+        data += _data
+        radio.pipe.write(b"\x06")
+        ack = radio.pipe.read(1)
+        if ack != b"\x06":
+            raise errors.RadioError("Radio refused block at %04x" % addr)
+
+        status = chirp_common.Status()
+        status.cur = addr
+        status.max = MAX_ADDR
+        status.msg = "Cloning from radio"
+        radio.status_fn(status)
+
+    radio.pipe.write(b"\x45")
+    return memmap.MemoryMapBytes(data)
+
+
+def do_upload(radio):
+    radio.pipe.parity = "E"
+    radio.pipe.timeout = 1
+    do_ident(radio)
+
+    mmap = radio._mmap
+    if callable(getattr(mmap, "get_byte_compatible", None)):
+        # for py3-CHIRP
+        mmap = mmap.get_byte_compatible()
+    for addr in range(0, MAX_ADDR, CHUNK_SIZE):
+        send(radio, make_frame(b"W", addr, CHUNK_SIZE, mmap[addr:addr + CHUNK_SIZE]))
+        ack = radio.pipe.read(1)
+        if ack != b"\x06":
+            raise errors.RadioError("Radio refused block at %04x" % addr)
+        radio.pipe.write(b"\x06")
+        ack = radio.pipe.read(1)
+        if ack != b"\x06":
+            raise errors.RadioError("Radio didn't ack our read ack")
+
+        status = chirp_common.Status()
+        status.cur = addr
+        status.max = MAX_ADDR
+        status.msg = "Cloning to radio"
+        radio.status_fn(status)
+
+    radio.pipe.write(b"\x45")
+
+
+def offset_for(freq):
+    for bounds, offset in zip(BOUNDS, OFFSETS):
+        if bounds[0] <= freq <= bounds[1]:
+            return offset
+    return 0
+
+
+class RadioSettingValueChannel(RadioSettingValueList):
+    """A setting that points to a defined channel."""
+    def __init__(self, radio, current_raw):
+        current = int(current_raw)
+        current_mem = radio.get_memory(current)
+        options = [
+            self._format_memory(mem)
+            for mem in [
+                radio.get_memory(n)
+                for n in range(*radio.get_features().memory_bounds)
+            ]
+        ]
```

Comment:
This is a nit, but my OCD won't allow me to avoid saying: I *hate* this style, as it's very vertically-wasteful, especially with the parens below.

I don't have very strict auto-style enforcement in chirp, so I shan't hold you up on this, but, I just had to say it :)

Reply:
this is the [psf/black](https://black.readthedocs.io/en/stable/) style that a lot of python projects are adopting these days so their maintainers don't have to think about formatting. lots of trailing commas, and lines with single parens and brackets.

i use it at work, it took some growing on me; and there are definitely a handful of coworkers that _hate_ it with a passion, and won't use it.

since getting used to it, I just tend to write stuff like this, even when I'm not using the formatter. 😬 


thanks for taking the change 👋 👍

Reply:
Yeah, I'm well aware of where it comes from, and I'm one of those who effing hate it :)